test/healthcheck_test.go

```diff
@@ -53,15 +53,51 @@ func replaceHealthCheckFunc(f func(context.Context, func() (interface{}, error),
 
 func newTestHealthServer() *testHealthServer {
 	return &testHealthServer{
-		update: make(chan struct{}, 1),
-		status: make(map[string]healthpb.HealthCheckResponse_ServingStatus),
+		watchFunc: defaultWatchFunc,
```

Comment:
Similar concept as in the balancer builder builder: delete this and call the other constructor with defaultWatchFunc as a parameter.